utils/helpers.ts

```
@@ -28,30 +28,81 @@ export function formatNumber(num: number | bigint | string): string {
-  const match = value.match(/^(\d+(\.\d+)?(?:e[+-]\d+)?)([ZEPTGMK])$/i);
+  if (!value) return BigInt(0);
+
+  // Match unit-suffixed values like "1.5M" or "2k" (supports scientific notation in the number)
+  const match = value.match(/^(\d+(?:\.\d+)?(?:e[+-]?\d+)?)([ZEPTGMK])$/i);
   if (match) {
-    const [, num, , unit] = match;
-    // Parse the number, which now handles scientific notation
-    const parsedNum = parseFloat(num);
+    const [, num, unit] = match;
+    const parsedNum = Number(num);
     const isoUnit = isoUnits.find((u) => u.iso.toUpperCase() === unit.toUpperCase()) || { threshold: 1, iso: '' };
```

ESLint

[error] Replace `(u)·=>·u.iso.toUpperCase()·===·unit.toUpperCase()` with `⏎······(u)·=>·u.iso.toUpperCase()·===·unit.toUpperCase()⏎····`

(prettier/prettier)